pkg/policy/api/egress.go

```diff
@@ -204,12 +204,24 @@ func (e *EgressRule) CreateDerivative() (*EgressRule, error) {
 		if err != nil {
 			return &EgressRule{}, err
 		}
-		if len(cidrSet) == 0 {
-			return &EgressRule{}, nil
+		if len(cidrSet) > 0 {
+			newRule.ToCIDRSet = append(e.ToCIDRSet, cidrSet...)
 		}
-		newRule.ToCIDRSet = append(e.ToCIDRSet, cidrSet...)
 	}
 	newRule.ToGroups = nil
+
+	for _, svc := range e.ToServices {
+		cidrSet := svc.GetCidrSet()
```

Comment:
I noticed that `svc.GetCidrSet` and `groups.GetCidrSet` use different code despite `svc.GetCidrSet` iterating over a list of providers (implying it can handle different types). I'm not sure where to put it, but a comment pointing this out might help to understand the code later.

Reply:
Oh, actually they are completely different... nevermind :/